libraries/Microsoft.Bot.Builder.Dialogs/Prompts/NumberPrompt.cs

```diff
@@ -152,5 +155,16 @@ protected override Task<PromptRecognizerResult<T>> OnRecognizeAsync(ITurnContext
 
             return Task.FromResult(result);
         }
+
+        private static List<ModelResult> RecognizeNumberWithUnit(string utterance, string culture)
+        {
+            var number = NumberRecognizer.RecognizeNumber(utterance, culture, NumberOptions.SuppressExtendedTypes);
+            var currency = NumberWithUnitRecognizer.RecognizeCurrency(utterance, culture);
+            var age = NumberWithUnitRecognizer.RecognizeAge(utterance, culture);
+            var temperature = NumberWithUnitRecognizer.RecognizeTemperature(utterance, culture);
+            var dimension = NumberWithUnitRecognizer.RecognizeDimension(utterance, culture);
+
+            return number.Any() ? number : currency.Any() ? currency : age.Any() ? age : temperature.Any() ? temperature : dimension;

```

Comment:
this looks really weird, I'd move to another function something like returnSomething and pass the parameters. Or change it to so some ifs clauses and put the value on another variable. It's not easy to read

Reply:
Did a refactor there